src/run.py

```diff
@@ -1,58 +1,132 @@
-#add 2 numbers
-def add(x, y):
+def add(x: float, y: float) -> float:
+    """
+    Adds 2 numbers together
+    Parameters
+    ---------
+    x: float
+    first number to add
+    
+    y: float
+    second number to add
+    
+    Returns
+    ---------
+    float
+    'x' added to 'y'. Mathematically equivalent to 'x + y'
+    """
     return x + y
 
-#subtract 2 numbers
-def subtract(x, y):
+def subtract(x: float, y: float) -> float:
+    """
+    Subtracts 2 numbers 
+    Parameters
+    ---------
+    x: float
+    first number to subtract
+    
+    y: float
+    second number to subtract
+    
+    Returns
+    ---------
+    float
+    'x' subtracted by 'y'. Mathematically equivalent to 'x - y'
+    """
     return x - y
 
-#multiply 2 numbers
-def multiply(x, y):
+def multiply(x: float, y: float) -> float:
+    """
+    Multiplies 2 numbers together
+    Parameters
+    ---------
+    x: float
+    first number to multiply
+    
+    y: float
+    second number to multiply
+    
+    Returns
+    ---------
+    float
+    'x' multiplied by 'y'. Mathematically equivalent to 'x * y'
+    """
     return x * y
 
-#divide 2 numbers
-def divide(x, y):
+def divide(x: float, y: float) -> float:
+    """
+    Divides 2 numbers 
+    Parameters
+    ---------
+    x: float
+    dividend
+    
+    y: float
+    divisor
+    
+    Returns
+    ---------
+    float
+    'x' divided by 'y'. Mathematically equivalent to 'x / y'
+    """
     return x / y
 
-previousCalc = 'no'
-
-print("Which operation?")
-print("1. Add")
-print("2. Subtract")
-print("3. Multiply")
-print("4. Divide")
-
-while True: 
-    #input operation
-    operation = input("Enter operation (1/2/3/4):")
-
-    #input validation
-    if operation in ('1', '2', '3', '4'):
-        try: 
-          if previousCalc == 'yes':
-            num1 = calc
-          else:
-            num1 = float(input("First number: "))
-          secondNum = float(input("Second number: "))
+# variable definitions
+prevCalc = 'no'
+num1 = 0
+num2 = 0
+
+# general start up
+def options():
+    print('Which operation?')
+    print('1. Add')
+    print('2. Subtract')
+    print('3. Multiply')
+    print('4. Divide')
+    
+
+while True:
+
+    # input operation
+    options()
+    choice = input('Choose which operation (1/2/3/4):')
+
+    # input validation & input numbers
+    if choice not in ('1', '2', '3', '4'):
+        print('Invalid input. Restart.')
+        break
+
+    if prevCalc == ('no', 'n'):
```

Comment:
It is now not asking for the first number:

```
Which operation?
1. Add
2. Subtract
3. Multiply
4. Divide
Choose which operation (1/2/3/4):
1
Second number: 
3
Your answer is  3.0
Another calc? (yes/no)
no
Would you like to use the previous answer? (yes/no)** Process Stopped **

Press Enter to exit terminal
```

Keep in mind `no` is not the same as `('no', 'n')`.